firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -4804,8 +4809,8 @@ extension BrowserViewController: TopTabsDelegate {
     }
 
     func topTabsDidPressNewTab(_ isPrivate: Bool) {
-        let shouldLoadCustomHomePage = isToolbarRefactorEnabled && NewTabAccessors.getHomePage(profile.prefs) == .homePage
-        let homePageURL = HomeButtonHomePageAccessors.getHomePage(profile.prefs)
+        let shouldLoadCustomHomePage = isToolbarRefactorEnabled && newTabSettings == .homePage
+        let homePageURL = NewTabHomePageAccessors.getHomePage(profile.prefs)
```

Comment:
This PR looks to be causing the https://github.com/mozilla-mobile/firefox-ios/issues/28117 based on running git bisect. 

This is the change that is causing the tests to fail. See more details in the comment on the [backport](https://github.com/mozilla-mobile/firefox-ios/pull/28081#discussion_r2225839515). We should address this for v142, but don't need to backport fix for v141.